### PR TITLE
Add sentry to origin-mobile

### PR DESCRIPTION
**Pull request by tomlinton**
Integrate Sentry with react-native. This is basically the output of the automated sentry integration script. I haven't attempted to test this at all, maybe you can try and spin it up @micahalcorn? There is an auth token you might need which goes in sentry.properties. You can ping me for it if you need it.

### Checklist:

- [] Test your work and double-check to confirm that you didn't break anything
- [ ] Wrap any displayed ETH addresses with [`formattedAddress`](https://github.com/OriginProtocol/origin/blob/master/origin-dapp/src/utils/user.js#L15-L17)
- [ ] Wrap any new text/strings for translation
- [ ] Run `npm run translations` if there are any changes to translated strings
- [ ] Map any new environment variables with a default value in the Webpack config
- [ ] Update any relevant READMEs and [docs](https://github.com/OriginProtocol/origin/tree/master/origin-docs)

### CODE REVIEW

File: origin-mobile/android/app/build.gradle [hunk 1]
```diff
@@ -148,6 +149,7 @@ android {
 }
 
 dependencies {
+    compile project(':react-native-sentry')
```
Comment (mikeshultz) [inline origin-mobile/android/app/build.gradle]: Instead of `compile`, this should be `implementation`.  It's just a warning right now but not for long.

Comment (tomlinton) [reply origin-mobile/android/app/build.gradle]: Would you mind updating this if you think it is a good idea @mikeshultz? I have no idea what I'm doing with android builds :grin:

**Comment by mikeshultz** [comment]
Made that change.  I'll give this a test build after either this or #1515 is merged, but eyeball parsing looks good.

**Comment by micahalcorn** [comment]
Thanks @tomlinton 🙏 This worked for me after following [this](https://github.com/facebook/react-native/issues/23183#issuecomment-458106496). Do I just add your `auth.token` to my local sentry.properties before building a version to submit to the App Store?

**Pull request merged**